main.js

```diff
@@ -1,186 +1,256 @@
-var App = {
-  localStorageKey: "r4find",
-  /* we can’t use localStorageKey here, because it’s undefined */
-  userEngines: JSON.parse(localStorage.getItem("r4find")) || {},
-
-	doEngines: {
-		r4: 'https://radio4000.com/add?url='
-	},
-	symbols: {
-		'!': {
-			name: 'search',
-			engines: {
-				a: 'https://www.amazon.com/gp/search?tag=internet4000-20&keywords=',
-				c: 'https://contacts.google.com/search/',
-				ciu: 'https://caniuse.com/#search=',
-				d: 'https://duckduckgo.com/?q=',
-				dd: 'https://devdocs.io/#q=',
-				dr: 'https://drive.google.com/drive/search?q=',
-				e: 'http://rover.ebay.com/rover/1/711-53200-19255-0/1?icep_ff3=9&pub=5575347480&toolid=10001&campid=5338215070&icep_sellerId=&icep_ex_kw=&icep_sortBy=12&icep_catId=&icep_minPrice=&icep_maxPrice=&ipn=psmain&icep_vectorid=229466&kwid=902099&mtid=824&kw=lg&icep_uq=',
-				g: 'https://encrypted.google.com/search?q=',
-				k: 'https://keep.google.com/?q=#search/text%3D',
-				l: 'https://www.linguee.com/search?query=',
-				lbc: 'https://www.leboncoin.fr/annonces/offres/ile_de_france/occasions/?th=1&q=',
-				lp: 'https://lpepfinder.com/#gsc.q=',
-				m: 'https://www.google.com/maps/search/',
-				osm: 'https://www.openstreetmap.org/search?query=',
-				r4: 'https://radio4000.com/search?search=',
-				tr: 'https://translate.google.com/?q=',
-				w: 'https://en.wikipedia.org/w/index.php?search=',
-				wa: 'http://www.wolframalpha.com/input/?i=',
-				y: 'https://www.youtube.com/results?search_query=',
-				'?': 'https://find.internet4000.com'
+(function (root, factory) {
+	if (typeof define === 'function' && define.amd) {
+		// AMD. Register as an anonymous module.
+		define([], factory);
+	} else if (typeof module === 'object' && module.exports) {
+		// Node. Does not work with strict CommonJS, but
+		// only CommonJS-like environments that support module.exports,
+		// like Node.
+		module.exports = factory();
+	} else {
+		// Browser globals (root is window)
+		root.Find = factory();
+	}
+}(typeof self !== 'undefined' ? self : this, function () {
+
+
+	// some private methods and variables;
+	var localStorageKey = 'i4find';
+
+	// generates new userSymbols from copying original symbols
+	// to be used with Find default symbols (Find.symbols)
+	var newUserSymbols = function() {
+		var fromSymbols = App.symbols;
+		var symbols = JSON.parse(JSON.stringify(fromSymbols))
+		Object.keys(symbols).forEach(symbol => {
+			symbols[symbol].engines = {}
+			if(symbol === '#') {
```

Comment:
Why delete this?

Reply:
I have been thinking about the security implication of having function as engines and user defined functions.

What do you think? Does it represent a possible vector?

```
https://find.internet4000.com/?q=#add test function(){console.log('hello')}()
```

Also the ux for adding a new function, `#`, is not really great.

Functions as engine is actually something I havent' had time to ponder, and would love a discussion with you about that.